### PR TITLE
glossary css

### PR DESCRIPTION
**Pull request by mruwnik**
This generally works, but might need a bit of css tweaking:
image

The main thing is what to do about the image size. This version just assumes it's up to the editors to get a version that looks good

### CODE REVIEW

File: app/components/Article/Contents.tsx [hunk 1]
```diff
@@ -125,8 +125,16 @@ const insertGlossary = (pageid: string, glossary: Glossary) => {
...
+          '<div class="glossary-popup flex-container">' +
+            '   <div class="contents">' +
+            `       <h3>${entry.term}</h3>` +
+            `       <div class="padding-bottom-24">${entry.contents}</div>` +
+            (entry.pageid
+              ? `<a href="/${entry.pageid}" class="button secondary">View full definition</a>`
+              : '') +
+            '   </div>' +
+            (entry.image ? `<img src="${entry.image}" alt="${entry.term}"/>` : '') +
+            '</div>'
```
Comment (Aprillion) [inline app/components/Article/Contents.tsx:137]: ah, have you considered 1 multiline string template, please? or maybe https://react.dev/reference/react-dom/server/renderToString if we need so much logic here 😅

File: app/components/Article/article.css [hunk 2]
```diff
@@ -77,6 +76,15 @@ article .link-popup::after {
   right: 100%;
 }
 
+article .link-popup .glossary-popup > .contents {
+  padding: var(--spacing-24) var(--spacing-40) var(--spacing-24);
```
Comment (Aprillion) [inline app/components/Article/article.css:80]: 2 or 4 values would make more sense here IMHO, but if the bottom is same as top, repeating it looks suspicious

File: app/components/Article/Contents.tsx [hunk 3]
```diff
@@ -125,8 +125,16 @@ const insertGlossary = (pageid: string, glossary: Glossary) => {
...
-            (entry.pageid ? `<br><a href="/${entry.pageid}">See more...</a>` : '')
+          '<div class="glossary-popup flex-container">' +
+            '   <div class="contents">' +
+            `       <h3>${entry.term}</h3>` +
+            `       <div class="padding-bottom-24">${entry.contents}</div>` +
+            (entry.pageid
+              ? `<a href="/${entry.pageid}" class="button secondary">View full definition</a>`
+              : '') +
+            '   </div>' +
+            (entry.image ? `<img src="${entry.image}" alt="${entry.term}"/>` : '') +
```
Comment (Aprillion) [inline app/components/Article/Contents.tsx:136]: `alt=""` for decorative images, please!! we shouldn't make users of screen reader listen to the same header twice 😅

File: app/components/Article/Contents.tsx [hunk 4]
```diff
@@ -79,6 +79,42 @@ const glossaryInjecter = (pageid: string, glossary: Glossary) => {
...
+  glossaryPopup.appendChild(contentsDiv)
+
+  if (image) {
+    const imageElem = document.createElement('img')
+    imageElem.src = image
+    glossaryPopup.appendChild(imageElem)
+  }
+
+  console.log(glossaryPopup.outerHTML)
+  return glossaryPopup.outerHTML
```
[the remark this answers is not in the document]
Comment (mruwnik) [reply app/components/Article/Contents.tsx:115]: oh... That kind :D give me a moment

[the remark this answers is not in the document]
Comment (mruwnik) [reply app/components/Article/Contents.tsx]: how about now?

[the remark this answers is not in the document]
Comment (Aprillion) [reply app/components/Article/Contents.tsx]: much better, thanks!

**Pull request merged**